subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -93,6 +93,9 @@ The embedded Kotlin has been updated from 1.8.20 to link:https://github.com/JetB
 The Kotlin language and API levels for the Kotlin DSL are still set to 1.8 for backwards compatibility.
 See the release notes for link:https://github.com/JetBrains/kotlin/releases/tag/v1.8.22[Kotlin 1.8.22] and link:https://github.com/JetBrains/kotlin/releases/tag/v1.8.21[Kotlin 1.8.21].
 
+Kotlin 1.9 dropped support for Kotlin language and API level 1.3.
+If you build Gradle plugins written in Kotlin with this version of Gradle and need to support Gradle <7.0 you need to stick to using the Kotlin Gradle Plugin <1.9.0 and configure the Kotlin language and API levels to 1.3.
```

Comment:
```suggestion
If you build Gradle plugins written in Kotlin with this version of Gradle and need to support Gradle <7.0 you need to stick to using the Kotlin Gradle Plugin <1.9.0 and configure the Kotlin language and API levels to 1.3. When using the `kotlin-dsl` plugin, you need to use version "4.0.14" or lower.
```

Reply:
As explained in https://github.com/gradle/gradle/issues/25868#issuecomment-1650063881, the workaround is unsupported, using previous versions of the `kotlin-dsl` plugin is also unsupported. That's why I omitted this from the upgrading guide on purpose.

Reply:
Makes sense!